### ISSUE TITLE
fixest v0.12.0: Error in cat_magic when running feols

### ISSUE THREAD

**Issue by mojihelen** [lrberge/fixest#490]
Trying feols for the first time using the example iris dataset from the Website. Received attached error message regarding cat_magic. Running R version 4.1.2 on Ubuntu server.

```
OLS estimation, Dep. Var.: Sepal.Length
Error: in cat_magic(..., .sep = .sep, .end = .end, .width = ...:
'string_magic' requires at least one character scalar to work.
Named arguments are only used as variables on which to apply interpolation.
FIX: please provide at least one non-named argument. 
```

Please advise how to resolve this, thanks!

<img width="385" alt="R_Error" src="https://github.com/lrberge/fixest/assets/30262533/b227e4d2-5ad7-4d7a-848b-8b6eceea5e5a">

**Comment by etiennebacher** [comment]
I can't reproduce this on R 4.3. Can you also show the version of `stringmagic` and `dreamerr` and update them if they are outdated?

**Comment by mojihelen** [comment]
Ubuntu Version 20.04.4 LTS
Versions of stringmagic and dreamerr seem ok
<img width="397" alt="Screenshot 2024-04-25 113546" src="https://github.com/lrberge/fixest/assets/30262533/236f21d7-92aa-4bd6-b3af-dd1f0f503549">

**Comment by lrberge** [comment]
Rhoooo, nooo. I'm sorry. Dependency hell pointing its nose. I'll try to fix asap.

**Comment by lrberge** [comment]
Hmm, I cannot replicate with R 4.0.2.

Can you run:
```R
library(dreamerr)
setDreamerr_show_stack(TRUE)
est = feols(Sepal.Length ~ Sepal.Width, iris)
print(est)
```

**Comment by mojihelen** [comment]
We have resolved the issue by upgrading to the latest R version:
<img width="320" alt="Screenshot 2024-04-25 192012" src="https://github.com/lrberge/fixest/assets/30262533/ada1204a-28ca-496a-8d6f-925ace6dd352">

Before that I had tried reinstalling the fixest package and I noticed there was a message regarding a previous package I had installed:
<img width="465" alt="test" src="https://github.com/lrberge/fixest/assets/30262533/c4acb6b0-eb7d-442a-ae6f-be2de2203fc7">

But deinstalling the package coin did not solve the issue..so not sure if it is related, but realized some packages previously installed must be causing the issue, thus decided to reinstall R.

Thank you so much for the quick help and support!

**Comment by lrberge** [comment]
The R namespace management is strict and the results should not depend on the loading of other packages (actually it can, but not for the reported problem).
I'm happy upgrading the R version solved the problem, although this is an inconvenience and far from optimal.

Since it's very likely that other users experience this problem following the new version of `fixest`, I'm leaving the issue open for information purposes.

Thanks for reporting.

**Comment by lrberge** [comment]
Solved, see #494 
Thanks again for swiftly reporting!
